packages/ra-tree-ui-materialui/src/DragPreview.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import compose from 'recompose/compose';
+import { withStyles } from '@material-ui/core/styles';
+import { translate } from 'ra-core';
+
+const styles = theme => ({
+    item: {
+        alignItems: 'center',
+        backgroundColor: theme.palette.action.active,
+        display: 'inline-flex',
+        height: 72,
+        minWidth: 72,
+        paddingTop: theme.spacing.unit * 1.5,
+        paddingBottom: theme.spacing.unit * 1.5,
+        paddingLeft: theme.spacing.unit * 6,
+        paddingRight: theme.spacing.unit * 4,
+    },
+});
+class DragPreview extends Component {
+    shouldComponentUpdate() {
+        return false;
+    }
+    render() {
+        const {
+            children,
+            className,
+            classes,
+            node,
+            style,
+            translate,
+        } = this.props;
+        return (
+            <div className={className || classes.item} style={style}>
+                {children
+                    ? typeof children === 'function'
+                        ? children({ node, translate })
+                        : children
+                    : translate('ra.tree.drag_preview', {
+                          _:
+                              'Node #%{id} |||| Node #%{id} with %{smart_count} children',
```

Comment:
well, that's the first time we do such a thing. I don't think we should put the default translation in the component. Makes me wonder how we should incorporate translations from optional packages. Any hint?

Reply:
Export addon translations packages as we do with the core ones and document how to merge them using lodash?

Reply:
I added an utility in `ra-core` for merging multiple language packages and added two language packages for `ra-tree`